lib/joint/RopeJoint.js

```diff
@@ -55,12 +55,18 @@ RopeJoint.prototype = create(RopeJoint._super.prototype);
  * Rope joint definition. This requires two body anchor points and a maximum
  * lengths. Note: by default the connected objects will not collide. see
  * collideConnected in JointDef.
- * 
+ *
  * @prop {float} maxLength The maximum length of the rope. Warning: this must be
  *       larger than linearSlop or the joint will have no effect.
+ *
+ * @prop {Vec2} def.localAnchorA The local anchor point relative to bodyA's origin.
+ * @prop {Vec2} def.localAnchorB The local anchor point relative to bodyB's origin.
  */
-var RopeJointDef = {
-  maxLength : 0.0
+
+var DEFAULTS = {
+  maxLength : 0.0,
+  localAnchorA : Vec2.zero(),
+  localAnchorB : Vec2.zero(),
```

Comment:
This seems a bit strange to me. You are requiring `localAnchorA` to be set in `RopeJointDef` but at the same time define default values. Is this intended? E.g. in `RevoluteJoint` there is no default value.

Reply:
I used default values to avoid failure if they are forgotten. But I checked box2d again, it also has default values, so probably they can actually be optional.